src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/MicrosoftNetCoreAnalyzersResources.resx

```diff
@@ -2129,4 +2129,13 @@ Widening and user defined conversions are not supported with generic types.</val
   <data name="UseStringMethodCharOverloadWithSingleCharactersTitle" xml:space="preserve">
     <value>Use char overload</value>
   </data>
+  <data name="AvoidSingleUseOfLocalJsonSerializerOptionsMessage" xml:space="preserve">
+    <value>Avoid single-use of local 'JsonSerializerOptions' as it is suboptimal, use a singleton instance instead to improve performance</value>
+  </data>
+  <data name="AvoidSingleUseOfLocalJsonSerializerOptionsDescription" xml:space="preserve">
+    <value>Avoid single-use of local 'JsonSerializerOptions' as it is suboptimal, use a singleton instance instead to improve performance.</value>
```

Comment:
NIT: the description could give more info than message, if its same the message then no need to add it (just use the message for description)

Reply:
Updated to "Reuse 'JsonSerializerOptions' instance by moving it to a static field, or use a singleton. Single-use 'JsonSerializerOptions' instances can substantially degrade the performance of your application."

Reply:
> Reuse 'JsonSerializerOptions' instance by moving it to a static field, or use a singleton

What's the difference between moving it to a static field and using a singleton?

Reply:
Yeah, I think singleton encompasses the former.

Reply:
Will update it to "Avoid creating a new 'JsonSerializerOptions' instance for every serialization operation. Cache and reuse instances instead. Single use 'JsonSerializerOptions' instances can substantially degrade the performance of your application."